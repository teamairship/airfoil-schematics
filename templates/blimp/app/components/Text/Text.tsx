import React, { FC } from 'react';
import { StyleProp, StyleSheet, Text as RNText, TextProps, TextStyle } from 'react-native';
import { colors, globalStyles, LINE_HEIGHT_MULTIPLIER } from '../../styles/globalStyles';

type Props = TextProps & {
  bold?: boolean;
  color?: string;
  fontSize?: number;
  italic?: boolean;
  style?: StyleProp<TextStyle>;
};

export const Text: FC<Props> = ({ bold, color, children, fontSize, italic, style }) => {
  return (
    <RNText
      allowFontScaling={false}
      style={[
        styles.standardText,
        fontSize ? { fontSize, lineHeight: fontSize * LINE_HEIGHT_MULTIPLIER } : null,
        italic ? styles.italic : null,
        bold ? styles.bold : null,
        color ? { color } : null,
        style ? style : null,
      ]}
    >
      {children}
    </RNText>
  );
};

const styles = StyleSheet.create({
  bold: { fontWeight: 'bold' },
  italic: { fontStyle: 'italic' },
  standardText: {
    ...globalStyles.standardText,
    color: colors.black,
  },
});
