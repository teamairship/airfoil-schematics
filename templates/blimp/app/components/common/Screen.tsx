import React from 'react';
import { StatusBar } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

interface Props {
  barStyle?: 'dark-content' | 'light-content';
}

export const Screen: React.FC<Props> = ({ barStyle = 'dark-content', children }) => {
  return (
    <>
      <StatusBar barStyle={barStyle} />
      <SafeAreaView style={{ flex: 1 }}>{children}</SafeAreaView>
    </>
  );
};
