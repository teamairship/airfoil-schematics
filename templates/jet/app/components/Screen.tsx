import React from 'react';
import { StatusBar, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

interface Props {
  barStyle?: 'dark-content' | 'light-content';
}

export const Screen: React.FC<Props> = ({ barStyle = 'dark-content', children }) => {
  return (
    <>
      <StatusBar barStyle={barStyle} />
      <SafeAreaView style={styles.safeAreaContainer}>{children}</SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
  safeAreaContainer: {
    flex: 1,
  },
});
