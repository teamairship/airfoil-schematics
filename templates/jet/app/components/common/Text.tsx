import React from 'react';
import { TextProps, TextStyle, StyleSheet, StyleProp } from 'react-native';
import { Text as ElementsText } from 'react-native-elements';

import { Color, colors, globalStyles, LINE_HEIGHT_MULTIPLIER } from '../../styles/globalStyles';

interface Props extends TextProps {
  color?: Color | string | null;
  bold?: boolean;
  italic?: boolean;
  marginBottom?: number;
  fontSize?: number | null;
  selectable?: boolean;
  style?: StyleProp<TextStyle>;
  testID?: string;
  numberOfLines?: number;
  ellipsizeMode?: 'head' | 'middle' | 'tail' | 'clip';
}

export const Text: React.FC<Props> = ({
  style,
  bold,
  color,
  italic,
  fontSize,
  numberOfLines,
  ellipsizeMode,
  children,
  testID,
}) => {
  return (
    <ElementsText
      style={[
        styles.standardText,
        fontSize ? { fontSize, lineHeight: fontSize * LINE_HEIGHT_MULTIPLIER } : null,
        italic ? styles.italic : null,
        bold ? styles.bold : null,
        color ? { color } : null,
        style,
      ]}
      testID={testID || ''}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {children}
    </ElementsText>
  );
};

const styles = StyleSheet.create({
  bold: {
    fontWeight: '800',
  },
  italic: {
    fontStyle: 'italic',
  },
  standardText: {
    ...globalStyles.standardText,
    color: colors.black,
  },
});
