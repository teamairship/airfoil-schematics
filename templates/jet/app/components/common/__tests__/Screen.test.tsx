import React from 'react';
import renderer from 'react-test-renderer';
import { Screen } from '../Screen';
import { Text } from '../Text';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Screen>
        <Text>Hello World!</Text>
      </Screen>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
