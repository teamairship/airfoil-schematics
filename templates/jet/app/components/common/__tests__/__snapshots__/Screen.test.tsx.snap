// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#121212",
        "fontSize": 16,
        "lineHeight": 27.2,
      }
    }
    testID=""
  >
    Hello World!
  </Text>
</RNCSafeAreaView>
`;
